use/wordcount-example.md

```diff
@@ -0,0 +1,413 @@
+
+The following sections explain these key concepts in detail, and break down the pipeline code into smaller sections.
+
+### Specifying Explicit DoFns
+
+When using `ParDo` transforms, you need to specify the processing operation that gets applied to each element in the input `PCollection`. This processing operation is a subclass of the SDK class `DoFn`. You can create the `DoFn` subclasses for each `ParDo` inline, as an anonymous inner class instance, as is done in the previous example (Minimal WordCount). However, it's often a good idea to define the `DoFn` at the global level, which makes it easier to unit test and can make the `ParDo` code more readable.
+
+In this example, `ExtractWordsFn` is a `DoFn` that is defined as a static class:
+
+```
+static class ExtractWordsFn extends DoFn<String, String> {
+    ...
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+        ...
+    }
+}
+```
+
+This `DoFn` (`ExtractWordsFn`) is the processing operation that `ParDo` applies to the `PCollection` of words:
+
+```
+PCollection<String> words = lines.apply(ParDo.of(new ExtractWordsFn()));
+```
+
+### Creating Composite Transforms
+
+If you have a processing operation that consists of multiple transforms or `ParDo` steps, you can create it as a subclass of `PTransform`. Creating a `PTransform` subclass allows you to create complex reusable transforms, can make your pipeline's structure more clear and modular, and makes unit testing easier.
+
+In this example, two transforms are encapsulated as the `PTransform` subclass `CountWords`. `CountWords` contains the `ParDo` that runs `ExtractWordsFn` and the SDK-provided `Count` transform.
+
+When `CountWords` is defined, we specify its ultimate input and output; the input is the `PCollection<String>` for the extraction operation, and the output is the `PCollection<KV<String, Long>>` produced by the count operation.
+
+```
+public static class CountWords extends PTransform<PCollection<String>,
+    PCollection<KV<String, Long>>> {
+  @Override
+  public PCollection<KV<String, Long>> apply(PCollection<String> lines) {
+
+    // Convert lines of text into individual words.
+    PCollection<String> words = lines.apply(
+        ParDo.of(new ExtractWordsFn()));
+
+    // Count the number of times each word occurs.
+    PCollection<KV<String, Long>> wordCounts =
+        words.apply(Count.<String>perElement());
+
+    return wordCounts;
+  }
+}
+
+public static void main(String[] args) throws IOException {
+  Pipeline p = ...
+
+  p.apply(...)
+   .apply(new CountWords())
+   ...
+}
+```
+
+### Using Parameterizable PipelineOptions
+
+You can hard-code various execution options when you run your pipeline. However, the more common way is to define your own configuration options via command-line argument parsing. 
+
+Add arguments to be processed by the command-line parser, and specify default values for them. You can then access the options values in your pipeline code.
```

Comment:
Point out explicitly that this makes the code more easily portable across different runners, as the runner and it's configuration can be specified at the commandline